### ISSUE TITLE
FyberMediationAdapter: NPE in onConfigurationReadyAndValid

### ISSUE THREAD

**Issue by too-far-away** [googleads/googleads-mobile-android-mediation#213]
After integrating Fyber adapter in AdMob mediation we observe application crashes in FyberMediationAdapter. It probably happens when initialization takes longer and does not finish by the time ads start showing.
The crash happens in `onConfigurationReadyAndValid` when the `completionCallback` passed in `waitForInitializationStatusAndReport` is **null**, although annotated as **non null**.
The null value for this parameter **does** get there in case the call chain starts (for example) with `requestBannerAd`, which calls `initializeFromBundle`, which in turn calls `initialize` passing null in the callback argument. Finally, `waitForInitializationStatusAndReport` adds configuration readiness listener with no callback and __sometimes__ this event fires twice, while the first time it was set up with the callback supposedly by AdMob initialization, and the second time after `requestBannerAd` was called and thus the callback is null there.
Sometimes, the event fires only once, and then everything works, so maybe the real issue is in the configuration manager.

**Comment by kapyar** [comment]
I know it might not help but Fyber is going to shut down on March. Dont think it will be priority for the team to fix issues for that provider

**Comment by not-there-yet** [comment]
I maybe wrong, but as I understand they close their current mediation platform, moving applications with DAU>25K to a new platform they call Fyber FairBid. Unless they are shutting down their business...

**Comment by mathieubouissiere** [comment]
@kapyar , Fyber is not shutting down. We are shutting down legacy products in March which is not related to the product mentioned by @not-there-yet. 
The product we are talking about here in this thread is **Fyber Marketplace** under AdMob mediation:
https://www.fyber.com/fyber-marketplace/ 
Regarding Fyber FairBid: https://www.fyber.com/meet-the-new-fyber-fairbid/, you are correct it's our new platform / mediation product. 
Just to clarify :)
Regarding the issue I will have a look and get back to you, we have also an email thread on which I will answer.

**Comment by foreza** [comment]
Thank you for filing this issue and the discussion!
We're currently working through our issue backlog and are closing older reports.

It's very likely that the NPE has been addressed since in a more recent version of the adapter: https://developers.google.com/admob/android/mediation/dt-exchange

As it has been over a year with no updates, I'm closing this issue. 

If you are still experiencing other problems with the Fyber (now DT exchange!) adapter or have questions, please **file a new issue**.
